File: Backend/controller/kas/metadata/metadataUpload.js
const metadata = require('../../../service/kas/metadata');

const metadataUpload = async function (req, res) {
    try {
        const upload = await metadata.kasMetadataUpload(req.body.metadata);

        if (upload['statusCode'] == 200) {
            res.status(200);
            res.json({statusCode: 200, data: upload['data']['result']});
        } else {
            const errorCode = upload['data']['code'];
            res.status(upload['statusCode']);

            switch (errorCode) {
                case 1174401:
                    res.json({statusCode: upload['statusCode'], msg: "입력한 데이터가 유효하지 않습니다."});
                    break;
                case 1174402:
                    res.json({statusCode: upload['statusCode'], msg: "JSON 포맷이 잘못되었습니다."});
                    break; 
                case 1174403:
                    res.json({statusCode: upload['statusCode'], msg: "메타데이터 JSON 포맷이 잘못되었습니다."});
                    break;
                case 1174404:
                    res.json({statusCode: upload['statusCode'], msg: "파일명이 잘못되었습니다."});
                    break;
                case 1174404:
                    res.json({statusCode: upload['statusCode'], msg: "중복된 파일명입니다."});
                    break;
                case 1175602:
                    res.json({statusCode: upload['statusCode'], msg: "데이터가 존재하지 않습니다."});
                    break;
                default:
                    res.json({statusCode: upload['statusCode'], msg: upload['data']['message']});
                    break;
            }
        }
    } catch (e) {
        console.error(e);
    }
}

module.exports = metadataUpload;

File: Etiquette/lib/Providers/KAS/Metadata/metadata_upload.dart
import 'package:dio/dio.dart';
import 'package:Etiquette/Models/dio_models.dart';
import 'package:Etiquette/Models/serverset.dart';

// 메타데이터 업로드
Future<Map<String, dynamic>> metadataUpload(Map<String, dynamic> metadata) async {
  Dio dio = getBasicDio();
  const url = "$SERVER_IP/kas/metadata/metadataUpload";
  try {
    final res = await dio.post(url, data: {
      "metadata": metadata,
    });
    Map<String, dynamic> data = res.data;
    return data;
    /*
    {
        statusCode: 200,
        data: {
            contentType: 'application/json',
            filename: 'c628a8b3-9875-1511-b781-ef9617b7e0de.json',
            uri: 'https://metadata-store.klaytnapi.com/bfc25e78-d5e2-2551-5471-3391b813e035/c628a8b3-9875-1511-b781-ef9617b7e0de.json'
        }
    }
    */
  } on DioError catch (e) {
    final handleError = e.response?.data;
    if (handleError == null) {
      Map<String, dynamic> data = {
        "statusCode": 400,
        "msg": "서버와의 연결이 원활하지 않습니다.",
      };
      return data;
    } else {
      Map<String, dynamic> data = {
        "statusCode": e.response?.statusCode,
        "msg": handleError["msg"],
      };
      return data;
    }
  } catch (ex) {
    Map<String, dynamic> data = {
      "statusCode": 400,
      "msg": ex.toString()
    };
    return data;
  }
}